shell.nix

```diff
@@ -0,0 +1,27 @@
+let
+  rust-overlay =
+    import (builtins.fetchGit {
+      url = "https://github.com/oxalica/rust-overlay.git";
+      rev = "84c58400556c1c5fa796cbc3215ba5bbd3bd848f";
+    });
+  nixpkgs = import <nixpkgs> { overlays = [ rust-overlay ]; };
+  rust-nightly = with nixpkgs; (rust-bin.selectLatestNightlyWith (toolchain: toolchain.default.override {
+    extensions = [ "rust-src" ];
```

Comment:
I think this is unnecessary

Reply:
I will try it

Reply:
It may be unnecessary for building. But [IDEs like it](https://github.com/paritytech/substrate/commit/6360780056d4cc885b8b066492c03ce7109ebe1d#diff-e53dfbfffe62ae3c0b411b3938ccffa9fb6a2ecc565f55785ef8daa756631a6b)

Reply:
So, is this an env setup or drml installer?

Reply:
And if this is an env setup the toolchain that you define here will be overwritten by the rust-toolchain.toml

Reply:
It is a  develop environment for nix user

Reply:
`rust-src` is unnecessary for building.

Reply:
I've already set up the tools for all the users

https://github.com/darwinia-network/darwinia-common/blob/main/rust-toolchain.toml#L3

Reply:
Not for nix, I think

Reply:
Once you run any command of the cargo.

The rust-toolchain.toml will overwrite the toolchain.

Reply:
What's the output of `rustup toolchain list`

Reply:
This is a setup to install cargo and any other related tools.

Reply:
Nix do not install rustup

Reply:
I think [rust-roolchain.toml](https://rust-lang.github.io/rustup/overrides.html) is for user who use rustup manage rust version.
And this is file is for nix user to set up a minimal develop environment.
There are two different ways for different users.